libsession/src/main/java/org/session/libsession/snode/utilities/PromiseUtil.kt

```diff
@@ -0,0 +1,13 @@
+package org.session.libsession.snode.utilities
+
+import nl.komponents.kovenant.Promise
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+suspend fun <T, E: Throwable> Promise<T, E>.await(): T {
+    return suspendCoroutine { cont ->
+        success { cont.resume(it) }
+        fail { cont.resumeWithException(it) }
```

Comment:
I think method references can be optimised further 🤔 

```suggestion
        success(cont::resume)
        fail(cont::resumeWithException)
```